cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/FileUtils.java

```diff
@@ -0,0 +1,119 @@
+            "YML"
+    ));
+
+    /**
+     *
+     * @param filePath  a relative file path beginning from the working directory
+     * @return          the content of the file as a String
+     * @throws IOException
+     */
+    public static String readLocalFile(String filePath) throws IOException {
+        checkNotNull(filePath);
+
+        String content = doReadLocalFile(filePath);
+
+        assert content != null;
+        return content;
+    }
+
+    private static String doReadLocalFile(String filePath) throws IOException {
+        File file = new File(filePath);
+        if (!isYamlFile(file.getName())) {
+            throw new InvalidFileTypeException("invalid or missing file extension: "
+                    + FilenameUtils.getExtension(file.getName()));
+        }
+        return readFile(file);
+    }
+
+    /**
+     *
+     * @param url   url to the host e.g. https://host.com/path/to/file.yml
+     * @return      the content of the file as a String
+     * @throws IOException
+     */
+    public static String readRemoteFile(String url) throws IOException, ProtocolException {
+        checkNotNull(url);
+
+        String content = doReadRemoteFile(url);
+
+        assert content != null;
+        return content;
+    }
+
+    private static String doReadRemoteFile(String url) throws IOException, ProtocolException {
+        URI uri = URI.create(url);
+
+        if (!isYamlFile(uri.getPath()) && !emptyFileExtension(uri.getPath())) {
+            throw new InvalidFileTypeException("invalid file extension: "
+                    + FilenameUtils.getExtension(uri.getPath()));
+        }
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+
+            try (CloseableHttpResponse response = httpClient.execute(new HttpGet(uri))) {
+
+                //TODO: more precise exception handling
+                if (response.getCode() != 200) {
+                    throw new HttpResponseException(response.getCode(), response.getReasonPhrase());
+                }
+
+                if (response.getHeader("Content-Type") != null
+                        && !response.getHeader("Content-Type").equals("text/plain")) {
+
+                    throw new HttpResponseException(response.getCode(),
+                            "invalid content type, was: " + response.getHeader("Content-Type"));
+                }
+
+                return EntityUtils.toString(response.getEntity());
+            }
+        }
+    }
+
+    public static boolean isYamlFile(String filename) {
+        return ALLOWED_FILE_EXTENSIONS.contains(FilenameUtils.getExtension(filename).toUpperCase());
+    }
+
+    public static boolean emptyFileExtension(String filename) {
+        return FilenameUtils.getExtension(filename).isEmpty();
+    }
+
+    private static String readFile(File file) throws IOException {
+        try (FileInputStream in = new FileInputStream(file)) {
+            return IOUtils.toString(in, StandardCharsets.UTF_8);
```

Comment:
We might not want to assume that every file is formatted in UTF-8.

Reply:
SnakeYaml supports loading from an InputStream object (see [here](https://bitbucket.org/asomov/snakeyaml/wiki/Documentation#markdown-header-loading-yaml)), so instead of returning the file contents as String, we could just return the InputStream and pass that to the yaml processor. The encoding would automatically be determined by the yaml library.

Reply:
Issue created: https://github.com/cf-control/cf-control/issues/20